src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/finalclass/Example1.java

```diff
@@ -0,0 +1,48 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="FinalClass"/>
+  </module>
+</module>
+*/
+
+// xdoc section -- start
+final class Example1 { // OK
+  private Example1() { }
+}
+
+class A { // violation, 'Class A should be declared as final.'
+  private A() { }
+}
+
+class B { // OK, since it has a public constructor
+  int field1;
+  String field2;
+  private B(int value) {
+    this.field1 = value;
+    this.field2 = " ";
+  }
+  public B(String value) {
+    this.field2 = value;
+    this.field1 = 0;
+  }
+}
+
+class TestAnonymousInnerClasses { // OK, 'class has an anonymous inner class.'
+  public static final TestAnonymousInnerClasses ONE
+          = new TestAnonymousInnerClasses() {};
+
+  private TestAnonymousInnerClasses() {
+  }
+}
+
+class Class1 {
+
+  private class Class2 { // violation, 'Class Class2 should be declared as final'
+  }
+
+  public class Class3 { // ok
```

Comment:
Aren't we trying to avoid `ok` comments without explanation? Same for the class on row 10.

Reply:
Would be nice to add a check for those too maybe? Not in this PR necessarily, just as a followup issue.

Reply:
https://github.com/checkstyle/checkstyle/pull/13556

Reply:
done

Reply:
Resolved.